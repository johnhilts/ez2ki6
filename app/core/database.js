export const firebaseConfig = {
  apiKey: 'AIzaSyDIoxipbDfUyTsZT_mCj-lShyNOfh_2ROg',
  authDomain: 'easynikkiroku.firebaseapp.com',
  databaseURL: 'https://easynikkiroku.firebaseio.com',
  storageBucket: '',
};

export function getUserRoot(owner) {
  return 'users/' + owner;
}
