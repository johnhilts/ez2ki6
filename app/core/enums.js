export const detailLevel = {day: 0, week: 1, month: 2, month_with_data: 3, };
