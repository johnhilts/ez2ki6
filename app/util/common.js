export const range  = (start, end) => {
  if (typeof(end) != "undefined") {
    return [...Array(end-start).keys()].map(v => start+v)
  }
  else {
    return [...Array(start).keys()]
  }
}
