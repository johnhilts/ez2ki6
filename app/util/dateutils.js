import moment from 'moment';

export const getCurrentFormattedYearMonth = () => {
  return moment().format('YYYYMM');
}

// Input: YYYYMM
// Output: YYYY-MM
const getReformattedYearMonth = (formattedYearMonth) => {
  return formattedYearMonth.substring(0, 4) + '-' + formattedYearMonth.substring(4);
}

// Input: YYYYMM, -1|1
// Output: YYYY-MM
export const getFormattedYearMonthByQueryYearMonth = (queryYearMonth, offset) => {
  let reformattedYearMonth = getReformattedYearMonth(queryYearMonth);
  return moment(reformattedYearMonth).add({months: offset}).format('YYYYMM');
}

// Input: YYYYMM
// Output: moment object
export const getYearMonthByFormattedYearMonth = (formattedYearMonth) => {
  let reformattedYearMonth = getReformattedYearMonth(formattedYearMonth);
  return moment(reformattedYearMonth);
}

// Input: y, m
// Output: YYYYMM
export const getFormattedYearMonthByYearMonth = (year, month) => {
  return moment(new Date(year, month, 1)).format('YYYYMM');
}

// Input: m
// Output: MMMM
export const getMonthName = (month) => {
  return moment(new Date(moment().year(), month, 1)).format('MMMM');
}

// returns date object based on default moment object
export const getCurrentDate = () => {
  return {year: moment().year(), month: moment().month(), day: moment().date(), };
}

// Input: y,m,d
// Output: true or false if y,m,d is for the actual current date
export const isToday = (year, month, day) => {
  let date = moment(new Date(year, month, day));
  let today = moment(new Date());
  return today.isSame(date, 'day');
}

// Input: y,m,d
// Output: date object
export const getDateFromYearMonthDay = (year, month, day) => {
  let date = moment(new Date(year, month, day));
  return {year: date.year(), month: date.month(), day: date.date(), };
}

// Input: y,m,d
// Output: MM/DD/YYYY
export const getFormattedMonthDayYear = (year, month, day) => {
  return moment(new Date(year, month, day)).format('MM/DD/YYYY');
}

export const getDaysByYearMonth = (year, month) => {
  return moment(new Date(year, month, 1)).daysInMonth();
}
