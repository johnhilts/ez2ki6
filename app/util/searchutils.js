export const searchByDateRange = (searchText, from, to, date) => {
    let fromDate = new Date(from.year, from.month, from.day);
    let toDate = new Date(to.year, to.month, to.day);
    let compareDate = new Date(date.year, date.month, date.day);
    return (
        date.dateInfo.toLowerCase().indexOf(searchText.toLowerCase()) >= 0
            && fromDate <= compareDate
            && toDate >= compareDate
    );
}