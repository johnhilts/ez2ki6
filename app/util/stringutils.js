import React from 'react';

export const formatLineBreaksForHtml = (str) => {
	return str.split("\n").map(function(item) {
		return (
			<span>
			{item}
			<br/>
			</span>
		       )
	});
}
