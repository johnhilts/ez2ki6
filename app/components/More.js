import React from 'react';
import { Link } from 'react-router'
import * as dateUtils from '../util/dateutils';
import * as themeUtils from '../util/themeutils';
import CalendarEntryContainer from '../containers/CalendarEntryContainer';

const searchResults = (day) => {
  return (
    <div key={day.key} className="col-sm-12">
      <Link to={{pathname: 'day', state: {monthInfo: day, }}}>{dateUtils.getFormattedMonthDayYear(day.year, day.month, day.day)} : {day.dateInfo}</Link>
    </div>
  )
}

const getCommonDateSelect = (currentElement, onDateChange, searchField, items) => {
  return (
    <select value={currentElement} onChange={onDateChange.bind(null, searchField)}>
      {items}
    </select>
  )
}

const getYearsSelect = (currentYear, years, onDateChange, searchField) => {
  const listOfYears = (year) => { return <option key={searchField + year} value={year}>{year}</option> }
    return (
      getCommonDateSelect(currentYear, onDateChange, searchField, years.map(listOfYears))
    )
}

const getMonthsSelect = (currentMonth, onDateChange, searchField) => {
  const getMonthRange = () => {
    let months = [];
    for (let m = 0; m < 12; m++) {
      months.push(m);
    }
    return months;
  }
  let months = getMonthRange();
  const listOfMonths = (month) => {return <option key={searchField + month} value={month}>{month+1}</option>}
  return (
    getCommonDateSelect(currentMonth, onDateChange, searchField, months.map(listOfMonths))
  )
}

const getDaysSelect = (currentYear, currentMonth, currentDay, onDateChange, searchField) => {
    const getMonthDayRange = (year, month) => {
      let range = [];
      let end = dateUtils.getDaysByYearMonth(currentYear, currentMonth);
      for (let i = 1; i <= end; i++) {
        range.push(i);
      }
      return range;
    }
    let days = getMonthDayRange(currentYear, currentMonth);
    const listOfDays = (day) => {return <option key={searchField + day} value={day}>{day}</option>}
      return (
        getCommonDateSelect(currentDay, onDateChange, searchField, days.map(listOfDays))
      )
    }

const RenderDateRanges = (props) => {
  const getDateSelect = (years, year, month, day, onDateChange, searchYear, searchMonth, searchDay) => {
    return (
      <span>
        {getMonthsSelect(month, onDateChange, searchMonth)}
          /
          {getDaysSelect(year, month, day, onDateChange, searchDay)}
          /
          {getYearsSelect(year, years, onDateChange, searchYear)}
      </span>
    )
  }
  let searchFields = props.searchFields;
  let fromDateSelect = getDateSelect(props.years, props.fromDate.year, props.fromDate.month, props.fromDate.day, props.onDateChange,
    searchFields.fromDateYear, searchFields.fromDateMonth, searchFields.fromDateDay);
  let toDateSelect = getDateSelect(props.years, props.toDate.year, props.toDate.month, props.toDate.day, props.onDateChange,
    searchFields.toDateYear, searchFields.toDateMonth, searchFields.toDateDay);

  return (
    <div className="row">
      <div>
        From:&nbsp;{fromDateSelect}
      </div>
      <div>&nbsp;</div>
      <div>
        To:&nbsp;&nbsp;{toDateSelect}
      </div>
    </div>
  )
}

const RenderThemeChoices = (props) => {
  let themes = themeUtils.themes;
  let selectedThemeId = props.user.themeId;
  return (
    <div>
      <label htmlFor="dark-theme">Dark Theme</label>&nbsp;<input type="radio" name="theme" id="dark-theme" value={themes.dark.id} onChange={props.onThemeChange} checked={selectedThemeId==themes.dark.id} />
      &nbsp;&nbsp;
      <label htmlFor="light-theme">Light Theme</label>&nbsp;<input type="radio" name="theme" id="light-theme" value={themes.light.id} onChange={props.onThemeChange} checked={selectedThemeId==themes.light.id} />
    </div>
  )
}

export default function More(props) {
  return (
    <div>
      <form onSubmit={props.onSearch}>
        <div className="row">
          <div className="col-sm-12">
            <input placeholder="Search" />&nbsp;<button type="submit"><span className="glyphicon glyphicon-search searchButtonText" aria-hidden="true"></span></button>
          </div>
        </div>
        <div>&nbsp;</div>
          <RenderDateRanges {...props} />
      </form>
      <div>&nbsp;</div>
      <div>
        <div className="row">
          {props.searchResults.map(searchResults)}
        </div>
      </div>
      <div>&nbsp;</div>
      <CalendarEntryContainer {...props} />
      <div>&nbsp;</div>
      <RenderThemeChoices {...props} />
    </div>
  )
}
