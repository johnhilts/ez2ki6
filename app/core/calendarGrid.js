import * as dateUtils from '../util/dateutils';
import * as enums from '../core/enums';

const saturday = 6;

export const getDayInfo = (dates, currentDate, isCurrentMonth, absoluteCellIndex) => {

  const matchesDate = dbDate => {return dbDate.year == currentDate.year && dbDate.month == currentDate.month && dbDate.day == currentDate.day;};

  let isToday = dateUtils.isToday(currentDate.year, currentDate.month, currentDate.day);
  let {year, month, day} = currentDate;
  let dayInfo = {year, month, day, absoluteIndex: absoluteCellIndex, isToday: isToday, isCurrentMonth: isCurrentMonth, };
  dayInfo.hasData = dates.some(matchesDate);
  if (dayInfo.hasData) {
    dayInfo.dateEntries = dates.filter(matchesDate);
  }

  return dayInfo;
}

export const getLastCalendarCellIndexFromStartOfCurrentMonth = (year, month, lastDay) => {
  let lastCurrentMonthWeekday = dateUtils.getWeekdayFromYearMonthDay(year, month, lastDay);
  return (
    lastCurrentMonthWeekday == saturday
    ? lastDay
    : lastDay + (saturday - lastCurrentMonthWeekday)
  )
}

export const buildMonthGrid = (dates, currentYearMonth) => {

  const getDayCellInfo = (currentDate, isCurrentMonth) => {
    return getDayInfo(dates, currentDate, isCurrentMonth, absoluteCellIndex++);
  }

  const daysInMonth = currentYearMonth.daysInMonth;
  let monthGrid = [];
  let absoluteCellIndex = 0;
  const currentYear = currentYearMonth.year;
  const currentMonth = currentYearMonth.month;
  const lastCalendarCellIndex = getLastCalendarCellIndexFromStartOfCurrentMonth(currentYear, currentMonth, daysInMonth);

  const processCalendarCell = (cellIndex) => {
    let day = cellIndex + 1;
    let date = dateUtils.getDateFromYearMonthDay(currentYear, currentMonth, day);

    const partOfWeekInPreviousMonth = (cellIndex) => {
      let previousMonthDate = dateUtils.addDays(currentYear, currentMonth, day, (weekday - cellIndex) * -1);
      return getDayCellInfo(previousMonthDate, false);
    }

    const weekInCurrentMonth = (cellIndex) => {
      return getDayCellInfo(date, true);
    }

    const partOfWeekInNextMonth = (cellIndex) => {
      return getDayCellInfo(date, false);
    }

    const daysInSameWeekPreviousMonth = (weekCellIndex) => {
      return day == 1 && weekday > weekCellIndex && weekCellIndex < saturday;
    }

    const daysInCurrentMonth = (weekCellIndex) => {
      return weekday == weekCellIndex && day <= daysInMonth;
    }

    const daysInSameWeekNextMonth = (weekCellIndex) => {
      return weekday == weekCellIndex && day > daysInMonth;
    }

    const weekLength = 7;
    let calendarCellsForWeek = [...Array(weekLength).keys()];
    let weekday = dateUtils.getWeekdayFromYearMonthDay(currentYear, currentMonth, day);

    monthGrid.push(...calendarCellsForWeek.filter(daysInSameWeekPreviousMonth).map(partOfWeekInPreviousMonth));
    monthGrid.push(...calendarCellsForWeek.filter(daysInCurrentMonth).map(weekInCurrentMonth));
    monthGrid.push(...calendarCellsForWeek.filter(daysInSameWeekNextMonth).map(partOfWeekInNextMonth));
  }

  let calendarCellsFromStartOfCurrentMonth = [...Array(lastCalendarCellIndex).keys()];
  calendarCellsFromStartOfCurrentMonth.map(processCalendarCell);

  return monthGrid;
}

export const setLinkDetailLevelForMonthView = (monthGrid) => {
  if (monthGrid.some(m => m.hasData)) {
    return enums.detailLevel.month_with_data
  }
  else {
    return enums.detailLevel.month
  }
}
