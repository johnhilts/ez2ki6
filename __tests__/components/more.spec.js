import React, { Component } from 'react';
import ReactTestRenderer from 'react-test-renderer';
import More from '../../app/components/More';
/*
import * as enums from '../../app/core/enums';
import * as calendarGrid from '../../app/core/calendarGrid';
import * as testHelper from '../../test/Helper/DateHelper';
*/
import * as dateUtils from '../../app/util/dateutils';

describe('More component snapshot test', () => {

  it('Should compare the More component with a snapshot', () => {
    // TODO - consider making MoreContainer.searchFields publicly accessible
    const searchFields = {searhText: 0, fromDateMonth: 1, fromDateDay: 2, fromDateYear: 3, toDateMonth: 4, toDateDay: 5, toDateYear: 6, };
    const component = ReactTestRenderer.create(
      <More
        searchResults={[]}
        fromDate={dateUtils.getCurrentDate()}
        toDate={dateUtils.getCurrentDate()}
        searchFields={searchFields}
        onDateChange={()=>{}}
        years={[dateUtils.getCurrentDate().year]}
        user={{themeId: 0}}
      />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })
});
