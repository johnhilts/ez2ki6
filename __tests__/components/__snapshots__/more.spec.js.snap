exports[`More component snapshot test Should compare the More component with a snapshot 1`] = `
<div>
  <form
    onSubmit={undefined}>
    <div
      className="row">
      <div
        className="col-sm-12">
        <input
          placeholder="Search" />
         
        <button
          type="submit">
          <span
            aria-hidden="true"
            className="glyphicon glyphicon-search invertedText" />
        </button>
      </div>
    </div>
    <div>
       
    </div>
    <div
      className="row">
      <div>
        From: 
        <span>
          <select
            onChange={[Function]}
            value={4}>
            <option
              value={0}>
              1
            </option>
            <option
              value={1}>
              2
            </option>
            <option
              value={2}>
              3
            </option>
            <option
              value={3}>
              4
            </option>
            <option
              value={4}>
              5
            </option>
            <option
              value={5}>
              6
            </option>
            <option
              value={6}>
              7
            </option>
            <option
              value={7}>
              8
            </option>
            <option
              value={8}>
              9
            </option>
            <option
              value={9}>
              10
            </option>
            <option
              value={10}>
              11
            </option>
            <option
              value={11}>
              12
            </option>
          </select>
          /
          <select
            onChange={[Function]}
            value={31}>
            <option
              value={1}>
              1
            </option>
            <option
              value={2}>
              2
            </option>
            <option
              value={3}>
              3
            </option>
            <option
              value={4}>
              4
            </option>
            <option
              value={5}>
              5
            </option>
            <option
              value={6}>
              6
            </option>
            <option
              value={7}>
              7
            </option>
            <option
              value={8}>
              8
            </option>
            <option
              value={9}>
              9
            </option>
            <option
              value={10}>
              10
            </option>
            <option
              value={11}>
              11
            </option>
            <option
              value={12}>
              12
            </option>
            <option
              value={13}>
              13
            </option>
            <option
              value={14}>
              14
            </option>
            <option
              value={15}>
              15
            </option>
            <option
              value={16}>
              16
            </option>
            <option
              value={17}>
              17
            </option>
            <option
              value={18}>
              18
            </option>
            <option
              value={19}>
              19
            </option>
            <option
              value={20}>
              20
            </option>
            <option
              value={21}>
              21
            </option>
            <option
              value={22}>
              22
            </option>
            <option
              value={23}>
              23
            </option>
            <option
              value={24}>
              24
            </option>
            <option
              value={25}>
              25
            </option>
            <option
              value={26}>
              26
            </option>
            <option
              value={27}>
              27
            </option>
            <option
              value={28}>
              28
            </option>
            <option
              value={29}>
              29
            </option>
            <option
              value={30}>
              30
            </option>
            <option
              value={31}>
              31
            </option>
          </select>
          /
          <select
            onChange={[Function]}
            value={2017}>
            <option
              value={2017}>
              2017
            </option>
          </select>
        </span>
      </div>
      <div>
         
      </div>
      <div>
        To:  
        <span>
          <select
            onChange={[Function]}
            value={4}>
            <option
              value={0}>
              1
            </option>
            <option
              value={1}>
              2
            </option>
            <option
              value={2}>
              3
            </option>
            <option
              value={3}>
              4
            </option>
            <option
              value={4}>
              5
            </option>
            <option
              value={5}>
              6
            </option>
            <option
              value={6}>
              7
            </option>
            <option
              value={7}>
              8
            </option>
            <option
              value={8}>
              9
            </option>
            <option
              value={9}>
              10
            </option>
            <option
              value={10}>
              11
            </option>
            <option
              value={11}>
              12
            </option>
          </select>
          /
          <select
            onChange={[Function]}
            value={31}>
            <option
              value={1}>
              1
            </option>
            <option
              value={2}>
              2
            </option>
            <option
              value={3}>
              3
            </option>
            <option
              value={4}>
              4
            </option>
            <option
              value={5}>
              5
            </option>
            <option
              value={6}>
              6
            </option>
            <option
              value={7}>
              7
            </option>
            <option
              value={8}>
              8
            </option>
            <option
              value={9}>
              9
            </option>
            <option
              value={10}>
              10
            </option>
            <option
              value={11}>
              11
            </option>
            <option
              value={12}>
              12
            </option>
            <option
              value={13}>
              13
            </option>
            <option
              value={14}>
              14
            </option>
            <option
              value={15}>
              15
            </option>
            <option
              value={16}>
              16
            </option>
            <option
              value={17}>
              17
            </option>
            <option
              value={18}>
              18
            </option>
            <option
              value={19}>
              19
            </option>
            <option
              value={20}>
              20
            </option>
            <option
              value={21}>
              21
            </option>
            <option
              value={22}>
              22
            </option>
            <option
              value={23}>
              23
            </option>
            <option
              value={24}>
              24
            </option>
            <option
              value={25}>
              25
            </option>
            <option
              value={26}>
              26
            </option>
            <option
              value={27}>
              27
            </option>
            <option
              value={28}>
              28
            </option>
            <option
              value={29}>
              29
            </option>
            <option
              value={30}>
              30
            </option>
            <option
              value={31}>
              31
            </option>
          </select>
          /
          <select
            onChange={[Function]}
            value={2017}>
            <option
              value={2017}>
              2017
            </option>
          </select>
        </span>
      </div>
    </div>
  </form>
  <div>
     
  </div>
  <div>
    <div
      className="row" />
  </div>
  <div>
     
  </div>
  <span>
     
  </span>
  <div>
     
  </div>
  <div>
    <label
      htmlFor="dark-theme">
      Dark Theme
    </label>
     
    <input
      checked={true}
      id="dark-theme"
      name="theme"
      onChange={undefined}
      type="radio"
      value={0} />
      
    <label
      htmlFor="light-theme">
      Light Theme
    </label>
     
    <input
      checked={false}
      id="light-theme"
      name="theme"
      onChange={undefined}
      type="radio"
      value={1} />
  </div>
</div>
`;
